lambda: support in return statement

### The following program `sample.gop` triggers an unexpected result

```coffee
func intSeq() func() int {
	i := 0
	return => {
		i++
		return i
	}
}
```


### Expected result

```console
compile passed
```


### Got

```console
panic: compileExpr failed: unknown - *ast.LambdaExpr2
```


### Gop Version

gop v1.1.0-alpha1-256-gf3285c6 darwin/amd64

### Additional Notes

_No response_

--- a/cl/compile_test.go
+++ b/cl/compile_test.go
@@ -3225,7 +3225,7 @@ func main() {
 `)
 }
 
-func _TestLambdaExpr3(t *testing.T) {
+func TestLambdaExpr3(t *testing.T) {
 	gopClTest(t, `
 func intSeq() func() int {
 	i := 0
@@ -3234,7 +3234,27 @@ func intSeq() func() int {
 		return i
 	}
 }
-`, `
+`, `package main
+
+func intSeq() func() int {
+	i := 0
+	return func() int {
+		i++
+		return i
+	}
+}
+`)
+	gopClTest(t, `
+func intDouble() func(int) int {
+	return i => i*2
+}
+`, `package main
+
+func intDouble() func(int) int {
+	return func(i int) int {
+		return i * 2
+	}
+}
 `)
 }
 

--- a/cl/error_msg_test.go
+++ b/cl/error_msg_test.go
@@ -130,6 +130,26 @@ var foo, foo1 func() = nil, => {}
 var foo func()
 _, foo = nil, => {}
 `)
+	codeErrorTest(t,
+		"./bar.gop:4:9: cannot use lambda expression as type int in return statement", `
+func intSeq() int {
+	i := 0
+	return => {
+		i++
+		return i
+	}
+}
+`)
+	codeErrorTest(t,
+		"./bar.gop:6:10: cannot use i (type int) as type string in return argument", `
+func intSeq() func() string {
+	i := 0
+	return => {
+		i++
+		return i
+	}
+}
+`)
 }
 
 func TestErrErrWrap(t *testing.T) {

--- a/cl/stmt.go
+++ b/cl/stmt.go
@@ -164,7 +164,18 @@ func compileReturnStmt(ctx *blockCtx, expr *ast.ReturnStmt) {
 					twoValue = (results.Len() == 2)
 				}
 			}
-			compileExpr(ctx, ret, twoValue)
+			switch v := ret.(type) {
+			case *ast.LambdaExpr, *ast.LambdaExpr2:
+				rtyp := ctx.cb.Func().Type().(*types.Signature).Results().At(i).Type()
+				sig, ok := rtyp.(*types.Signature)
+				if !ok {
+					panic(ctx.newCodeErrorf(
+						ret.Pos(), "cannot use lambda expression as type %v in return statement", rtyp))
+				}
+				compileLambda(ctx, v, sig)
+			default:
+				compileExpr(ctx, ret, twoValue)
+			}
 		}
 	}
 	ctx.cb.Return(len(expr.Results), expr)

